### ISSUE TITLE
network: bridge netdev VLAN protocol 802.1ad - Protocol not supported

### ISSUE THREAD

**Issue by jloeser** [systemd/systemd#22469]
**systemd version the issue has been seen with**
``` 
# systemctl --version 
systemd 246 (246.16+suse.191.g3850086c65)
+PAM +AUDIT +SELINUX -IMA +APPARMOR -SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +
ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```

**Used distribution**
> openSUSE Leap 15.3

**Linux kernel version used** (`uname -a`)
 > `Linux localhost 5.3.18-59.27-default #1 SMP Tue Oct 5 10:00:40 UTC 2021 (7df2404) x86_64 x86_64 x86_64 GN
U/Linux`

**CPU architecture issue was seen on**
 > x86_64

**Expected behaviour you didn't see**

I want to create a bridge interface with VLAN filtering protocol 802.1ad (Q-in-Q). According to the docs (`man systemd.netdev`) this should work with the following configuration:

```
localhost:~ # cat /etc/systemd/network/br0.netdev 
[NetDev]
Name=br0
Kind=bridge

[Bridge]
VLANFiltering=yes
VLANProtocol=802.1ad
localhost:~ # 
localhost:~ # networkctl reload
localhost:~ # networkctl up br0
```

Result *should* be:
```
localhost:~ # ip -d l show br0 | grep -oiE "vlan_filtering (0|1) vlan_protocol (802.1ad|802.1q)"
vlan_filtering 1 vlan_protocol 802.1ad
```

**Unexpected behaviour you saw**

After reloading configuration and bringing up the interface it ends up with this:
```
localhost:~ # ip -d l show br0 | grep -oiE "vlan_filtering (0|1) vlan_protocol (802.1ad|802.1q)"
vlan_filtering 0 vlan_protocol 802.1Q
```

...and in the logs I see (after reloading, see below):
```
Feb 09 18:32:51 localhost systemd-networkd[872]: br0: Bridge parameters could not be set: Protocol not supported
```

**Steps to reproduce the problem**
```
localhost:~ # cat > /etc/systemd/network/br0.netdev <<EOF
[NetDev]
Name=br0
Kind=bridge

[Bridge]
VLANFiltering=yes
VLANProtocol=802.1ad
EOF
localhost:~ # networkctl reload
localhost:~ # networkctl up br0
localhost:~ # ip -d l show br0 | grep -oiE "vlan_filtering (0|1) vlan_protocol (802.1ad|802.1q)"
```

Adding an bridge device with desired configuration via ```ip``` on the same machine works:
```
localhost:~ # ip link add br1 type bridge vlan_filtering 1 vlan_protocol 802.1ad
localhost:~ # ip -d l show br1 | grep -oiE "vlan_filtering (0|1) vlan_protocol (802.1ad|802.1q)"
vlan_filtering 1 vlan_protocol 802.1ad
```

**Additional program output to the terminal or log subsystem illustrating the issue**

```
localhost:~ # journalctl -f | grep br0
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: loaded bridge
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Creating
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: New device has no master, continuing without
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Flags change: +MULTICAST +BROADCAST
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Link 7 added
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: link pending udev initialization...
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: netdev has index 7
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: netdev ready
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Saved original MTU: 1500
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Created
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Bridge parameters could not be set: Protocol not supported
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: rtnl: received neighbor message with invalid family '7', ignoring.
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: rtnl: received neighbor message with invalid family '7', ignoring.
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: rtnl: received neighbor message with invalid family '7', ignoring.
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Gained new MAC address: 96:a2:d3:53:52:47
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: udev initialized link
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: State changed: pending -> initialized
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: Link state is up-to-date
Feb 09 18:29:05 localhost systemd-networkd[872]: br0: State changed: initialized -> unmanaged
Feb 09 18:29:15 localhost systemd-networkd[872]: br0: Flags change: +UP
Feb 09 18:29:15 localhost systemd-networkd[872]: br0: Link UP
localhost:~ # lsmod | grep 8021q
8021q                  36864  0
garp                   16384  1 8021q
mrp                    20480  1 8021q

```

**Comment by yuwata** [comment]
Sorry, v246 is too old here. Could you check the issue is reproducible with a recent release (v249 or v250) or current git HEAD?

**Comment by jloeser** [comment]
Sure. Trying with HEAD:

```
localhost:~/systemd # PAGER= git log --oneline -1 
19ff06b3a4 (HEAD -> main, origin/main, origin/HEAD) udev/net: support Match.Firmware= in .link files (#22462)
localhost:~/systemd # meson setup build/ && meson compile -C build/
localhost:~/systemd # DESTDIR=/usr meson install -C build/
```

Reboot

```
localhost:~ # systemctl --version 
systemd 250 (250-814-g0628d48)
+PAM -AUDIT +SELINUX +APPARMOR +IMA +SMACK -SECCOMP -GCRYPT -GNUTLS +OPENSSL -ACL +BLKID -CURL -ELFUTILS -FIDO2 -IDN2 -IDN -IPTC -KMOD -LIBCRYPTSETUP -LIBFDISK -PCRE2 -PWQUALITY -P11KIT -QRENCODE -BZIP2 -LZ4 -XZ +ZLIB -ZSTD -BPF_FRAMEWORK -XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified
localhost:~ # cat /etc/systemd/network/br0.netdev
[NetDev]
Name=br0
Kind=bridge

[Bridge]
VLANFiltering=yes
VLANProtocol=802.1ad
localhost:~ # ip -d l show br0 | grep -oiE "vlan_filtering (0|1) vlan_protocol (802.1ad|802.1q)"
vlan_filtering 0 vlan_protocol 802.1Q
```

Same in the logs:
```
Feb 10 08:22:27 localhost systemd-networkd[873]: br0: Bridge parameters could not be set: Protocol not supported
```

**Comment by jloeser** [comment]
Also tried with latest openSUSE Tumbleweed image:

```
localhost:~ # systemctl --version 
systemd 249 (249.9+suse.82.g117bd7f14a)
+PAM +AUDIT +SELINUX +APPARMOR -IMA -SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified
localhost:~ # cat /etc/systemd/network/br0.netdev
[NetDev]
Name=br0
Kind=bridge

[Bridge]
VLANFiltering=yes
VLANProtocol=802.1ad
localhost:~ # ip -d l show br0 | grep -oiE "vlan_filtering (0|1) vlan_protocol (802.1ad|802.1q)"
vlan_filtering 0 vlan_protocol 802.1Q
```

**Comment by yuwata** [comment]
Thanks. I will take a look.

**Comment by yuwata** [comment]
Fix is waiting in #22471.
